tests/test_models.py

```diff
@@ -149,31 +160,129 @@ def test_serialize_a_customer(self):
         """It should not deserialize bad data"""
         data = "this is not a dictionary"
         customer = Customer()
         self.assertRaises(DataValidationError, customer.deserialize, data)
 
+    def test_deserialize_address_key_error(self):
+        """It should not Deserialize an address with a KeyError"""
+        address = Address()
+        self.assertRaises(DataValidationError, address.deserialize, {})
+
+    def test_deserialize_address_type_error(self):
+        """It should not Deserialize an address with a TypeError"""
+        address = Address()
+        self.assertRaises(DataValidationError, address.deserialize, [])
+
+    def test_add_customer_address(self):
+        """It should Create an Customer with an Address and add it to the database"""
+        customers = Customer.all()
+        self.assertEqual(customers, [])
+        customer = CustomerFactory()
+        address = AddressFactory(customer=customer)
+        customer.addresses.append(address)
+        customer.create()
+        address.create()
+        # Assert that it was assigned an id and shows up in the database
+        self.assertIsNotNone(customer.id)
+        customers = Customer.all()
+        self.assertEqual(len(customers), 1)
+
+        new_customer = customer.find(customer.id)
+        self.assertEqual(new_customer.addresses[0].street, address.street)
+
+        address2 = AddressFactory(customer=customer)
+        customer.addresses.append(address2)
+        customer.update()
+
+        new_customer = Customer.find(customer.id)
+        self.assertEqual(len(new_customer.addresses), 2)
+        self.assertEqual(new_customer.addresses[1].street, address2.street)
+
+    def test_update_customer_address(self):
+        """It should Update an Customer's address"""
+        customers = Customer.all()
+        self.assertEqual(customers, [])
+
+        customer = CustomerFactory()
+        address = AddressFactory(customer=customer)
+        customer.create()
+
+        # Assert that it was assigned an id and shows up in the database
+        self.assertIsNotNone(customer.id)
+        customers = Customer.all()
+        self.assertEqual(len(customers), 1)
+
+        # Fetch it back
+        customer = Customer.find(customer.id)
+        old_address = customer.addresses[0]
+        print("%r", old_address)
+        self.assertEqual(old_address.city, address.city)
+        # Change the city
+        old_address.city = "XX"
+        customer.update()
+        address.update()
+
+        # Fetch it back again
+        customer = Customer.find(customer.id)
+        address = customer.addresses[0]
+        self.assertEqual(address.city, "XX")
+
+    def test_delete_customer_address(self):
+        """It should Delete a Customer's address"""
+        customers = Customer.all()
+        self.assertEqual(customers, [])
+
+        customer = CustomerFactory()
+        address = AddressFactory(customer=customer)
+        customer.create()
```

Comment:
customer.addresses.append(address) didnt happen here. am i missing something?

Reply:
Makes sense, added it.